nf_core/rocrate.py

```diff
@@ -0,0 +1,300 @@
+            self.crate,
+            "#nextflow",
+            properties={
+                "@type": ["ComputerLanguage", "SoftwareApplication"],
+                "name": "Nextflow",
+                "url": "https://www.nextflow.io/",
+                "identifier": "https://www.nextflow.io/",
+                "version": self.pipeline_obj.nf_config.get("manifest.nextflowVersion", ""),
+            },
+        )
+        self.crate.add(programming_language)
+
+        # Conform to RO-Crate 1.1 and workflowhub-ro-crate
+        self.crate.update_jsonld(
+            {
+                "@id": "ro-crate-metadata.json",
+                "conformsTo": [
+                    {"@id": "https://w3id.org/ro/crate/1.1"},
+                    {"@id": "https://w3id.org/workflowhub/workflow-ro-crate/1.0"},
+                ],
+            }
+        )
+
+        # Set main entity file
+        self.set_main_entity("main.nf")
+
+        # add readme as description
+        readme = Path("README.md")
+
+        try:
+            self.crate.description = readme.read_text()
+        except FileNotFoundError:
+            log.error(f"Could not find README.md in {self.pipeline_dir}")
+
+        self.crate.license = "MIT"
+
+        # add doi as identifier
+        self.crate.name = f'Research Object Crate for {self.pipeline_obj.nf_config.get("manifest.name")}'
+
+        if "dev" in self.pipeline_obj.nf_config.get("manifest.version", ""):
+            self.crate.CreativeWorkStatus = "InProgress"
+        else:
+            self.crate.CreativeWorkStatus = "Stable"
+
+        # Add all other files
+        self.add_workflow_files()
+
+    def set_main_entity(self, main_entity_filename: str):
+        """
+        Set the main.nf as the main entity of the crate and add necessary metadata
+        """
+
+        wf_file = self.crate.add_jsonld(
+            {
+                "@id": main_entity_filename,
+                "@type": ["File", "SoftwareSourceCode", "ComputationalWorkflow"],
+            },
+        )
+        self.crate.mainEntity = wf_file
+        self.add_main_authors(wf_file)
+        wf_file.append_to("programmingLanguage", {"@id": "#nextflow"})
+        # get keywords from nf-core website
+        remote_workflows = requests.get("https://nf-co.re/pipelines.json").json()["remote_workflows"]
+        # go through all remote workflows and find the one that matches the pipeline name
+        topics = ["nf-core", "nextflow"]
+        for remote_wf in remote_workflows:
+            if remote_wf["name"] == self.pipeline_obj.pipeline_name.replace("nf-core/", ""):
+                topics = topics + remote_wf["topics"]
+                break
+
+        log.debug(f"Adding topics: {topics}")
+        wf_file.append_to("keywords", topics)
+
+    def add_main_authors(self, wf_file):
+        """
+        Add workflow authors to the crate
+        NB: We don't have much metadata here - scope to improve in the future
```

Comment:
Had some discussion about revisiting the `CITATIONS.cff` file and other things. For example, in theory I think that we can even associate ORCiD identifiers here and all sorts?

It'd also be cool if we could scrape GitHub contributors or something 🤔

Reply:
Nooo way, you did ORCiD lookups?? haha, amazing 🙇🏻

Reply:
added now all contributors to "main.nf" according to git (if they have an "identifiable" name (i.e. a space in it)